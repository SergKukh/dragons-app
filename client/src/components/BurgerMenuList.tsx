import { FC, useRef } from 'react';
import { useNavigate } from 'react-router-dom';
import { useAppSelector } from '../hooks/redux';
import { RouteNames } from '../router';
import styles from './styles/BurgerMenuList.module.css';

export interface ListItem {
    title: string,
    handler: () => void
}

interface BurgerMenuListProps {
    isActive: boolean,
    items: ListItem[],
    closeHandler: () => void
}

const BurgerMenuList: FC<BurgerMenuListProps> = ({ isActive, items, closeHandler }) => {
    const hide = useRef<boolean>(false);
    const { user } = useAppSelector(state => state.authReducer);
    const navigate = useNavigate();

    if (isActive && !hide.current) {
        hide.current = true;
    }

    const profileHandler = () => {
        navigate(RouteNames.PROFILE);
        closeHandler();
    }

    return (
        <div className={`${styles.container} ${isActive ? styles.active : ''} ${hide.current ? styles.hide : ''}`}>
            {items.map(item =>
                <div className={styles.item}
                    onClick={item.handler}
                    key={item.title}
                    data-testid='burger_menu_item'
                >
                    {item.title}
                </div>)}
            <div className={styles.user}>
                <div onClick={profileHandler}>{user.email}</div>
                {!user.isActivated && <div className={styles.red}>{'not activated'}</div>}
            </div>
        </div>
    );
};

export default BurgerMenuList;