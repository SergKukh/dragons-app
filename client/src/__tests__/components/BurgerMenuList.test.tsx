import { render, screen } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import BurgerMenuList, { ListItem } from "../../components/BurgerMenuList";

jest.mock('../../hooks/redux', () => ({
    useAppSelector: () => ({
        user: {
            email: 'test@test.com',
            isActivated: true,
            id: '1'
        }
    })
}))

jest.mock('react-router-dom', () => ({
    useNavigate: () => (route: string) => { }
}))

describe('burger menu', () => {
    test('items', async () => {
        const items: ListItem[] = [
            { title: '1', handler: () => { } },
            { title: '2', handler: () => { } },
            { title: '3', handler: () => { } },
        ]
        await act(async () => {
            render(<BurgerMenuList isActive={false} items={items} closeHandler={() => { }} />);
        })
        const elems = await screen.findAllByTestId('burger_menu_item');
        expect(elems.length).toBe(3);
        expect(elems).toMatchSnapshot();
    })
})