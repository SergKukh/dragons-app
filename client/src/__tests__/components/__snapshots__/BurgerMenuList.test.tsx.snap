// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`burger menu items 1`] = `
Array [
  <div
    class="item"
    data-testid="burger_menu_item"
  >
    1
  </div>,
  <div
    class="item"
    data-testid="burger_menu_item"
  >
    2
  </div>,
  <div
    class="item"
    data-testid="burger_menu_item"
  >
    3
  </div>,
]
`;
